newIDE/app/src/InstancesEditor/InstancesRenderer/LayerRenderer.js

```diff
@@ -379,19 +381,41 @@ export default class LayerRenderer {
       renderedInstance._pixiObject.eventMode = 'static';
       panable(renderedInstance._pixiObject);
       makeDoubleClickable(renderedInstance._pixiObject);
+      renderedInstance._pixiObject.addEventListener('mousemove', event => {
+        console.log('mousemove');
+        if (
+          shouldPreventRenderingInstanceEditors() ||
+          !isMouseOnSceneEditorCanvas({ x: event.clientX, y: event.clientY })
+        ) {
+          this.onOutInstance(instance);
+          return;
+        }
+      });
       renderedInstance._pixiObject.addEventListener('click', event => {
+        console.log('click');
         if (event.data.originalEvent.button === 0)
           this.onInstanceClicked(instance);
       });
       renderedInstance._pixiObject.addEventListener('doubleclick', () => {
+        console.log('double click');
         this.onInstanceDoubleClicked(instance);
       });
-      renderedInstance._pixiObject.addEventListener('mouseover', () => {
+      renderedInstance._pixiObject.addEventListener('mouseover', event => {
+        console.log('mouseover', event);
+        if (
+          shouldPreventRenderingInstanceEditors() ||
```

Comment:
I don't think it's reasonable to check on each frame if there's dialog open (DOM query). I think it should at least be cached for the current frame.